validator/index.js

```diff
@@ -266,21 +460,41 @@ function main() {
 
   program.command('* <fileOrUrlOrMinus...>')
       .description('Validates list of files or urls (default).')
-      .action((validateOrUrlOrMinus) => {
-        if (validateOrUrlOrMinus.length == 0) {
+      .action((fileOrUrlOrMinus) => {
+        if (fileOrUrlOrMinus.length == 0) {
           program.outputHelp();
           process.exit(1);
         }
-        readFileOrDownload(
-            program.validator_js,
-            (validatorScript) => {
-              vm.runInThisContext(validatorScript);
-              validateFiles(validateOrUrlOrMinus);
-            },
-            (errorMessage) => {
-              console.error('Could not fetch validator.js: ' + errorMessage);
-              process.exitCode = 1;
-            });
+        for (const item of fileOrUrlOrMinus) {
+          let input;
+          if (item === '-') {
+            input = readFromStdin();
+          } else if (isHttpOrHttpsUrl(item)) {
+            input = readFromUrl(item);
+          } else {
+            input = readFromFile(item);
+          }
+          input
+              .then((data) => {
+                getInstance(program.validator_js)
+                    .then((validator) => {
+                      const validationResult = validator.validateString(data);
+                      logValidationResult(item, validationResult);
+                      if (validationResult.status !== 'PASS') {
+                        process.exitCode = 1;
+                      }
+                    })
+                    .catch((error) => {
+                      console.error(
```

Comment:
Not sure I understand.
So this console.error here is regarding the getInstance(program.validator_js) promise. There is no stdin involved here. If this promise fails with an error, it's always because something went wrong with fetching / instantiating the validator. Hence, this error logging points out that something went wrong with the validator.js fetching, and which error message. If I just emit the error, then the context is lost. It may then tell me something like ENOENT or HTTP 404 or whatever, but may not necessarily include the name of the file.
As for stdin etc., if something went wrong with the input this would be handled with line 494. Also in that case, it's important to add the context.
In my testing, these errors looked reasonable to me, but there's a good chance that something better can be done.  I'd like to produce descriptive, useful error messages for command line tool users here (e.g., no stack traces or things where one has to look into the code to understand what went wrong). Whatever way best accomplishes this I'd be super happy to do.